python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,567 @@
+        """If monochromatic flats, check that laser can be enabled,
+        check temperature, turn on laser to warm up.
+        move linearstage_select to correct location for
+        Mono or Whitelight flats
+        """
+        config_data = self.get_calibration_configuration(sequence_name)
+
+        calibration_type = getattr(CalibrationType, str(config_data["calib_type"]))
+
+        if calibration_type == CalibrationType.WhiteLight:
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_led_location
+            )
+        else:
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_laser_location
+            )
+            await self.laser_setup(config_data["laser_mode"])
+            await self.rem.tunablelaser.cmd_startPropagating.set_start()
+
+    async def laser_setup(self, mode: str) -> None:
+        """Is the laser on? what is the temp like? thermal control on?"""
+        # check the thermal system is running
+        # check set temp
+        # check actual temp and determine if good enough
+        # check the interlockState
+
+        if mode == TunableLaser.Mode.CONTINUOUS:
+            await self.rem.tunablelaser.cmd_setContinuousMode.set_start()
+        elif (mode == TunableLaser.Mode.BURST) | (mode == TunableLaser.Mode.TRIGGER):
+            await self.rem.tunablelaser.cmd_setBurstMode.set_start()
+
+    async def prepare_for_flat(self, config_name: str) -> None:
+        """Configure the ATMonochromator according to the flat parameters
+
+        Parameters
+        ----------
+        config_name : `str`
+            name of the type of configuration you will run, which is saved
+            in the configuration.yaml files
+
+        Raises
+        ------
+        RuntimeError:
+
+        """
+        config_data = self.get_calibration_configuration(config_name)
+
+        calibration_type = getattr(CalibrationType, str(config_data["calib_type"]))
+        filter_name = config_data[
+            "filter"
+        ]  # not sure if this is an enumeration somewhere?
+
+        task_setup_camera = (
+            self.mtcamera.setup_instrument(
+                filter=config_data["atspec_filter"],
+                grating=config_data["atspec_grating"],
+            )
+            if self.mtcamera is not None and config_data["use_camera"]
+            else utils.make_done_future()
+        )
+        if self.mtcamera is None and config_data["use_camera"]:
+            raise RuntimeError(
+                f"MTCamera is not defined but {config_name} requires it. "
+                "Make sure you are instantiating ComCam and passing it to MTCalsys."
+            )
+
+        if calibration_type == CalibrationType.WhiteLight:
+            leds_ = self.led_per_filter[filter_name]
+            task_select_wavelength = (
+                self.linearstage_led_select.cmd_moveAbsolute.set_start(
```

Comment:
shouldn't this use the change_wavelength function instead?

Reply:
It's very different with the LED - you move a stage for a given filter. I don't think putting those two things in the same function makes sense. But perhaps I change the name of `change_wavelength` to `change_laser_wavelength`

Reply:
Do you mean the MTCalsys has 2 ways to change the wavelength? we can the laser wavelength or the LEDs? 

If that is the case, we have 2 options:

1. We can pick one (probably the most commonly used one) to be encapsulated by the `change_wavelength` method and create another method to encapsulate the other operation (e.g. `change_laser_wavelength`).

or

2. We introduce a parameter to the `change_wavelength` method that would allow us to pass what type of "operation" we want to do. It could be something like this"

```
async def change_wavelength(self, wavelength: float, type: CalibrationType) -> None:
    ...
```

I am assuming above that you would be able to determine what do to from the calibration type, but this could be a new enumeration. We would also have to figure out what to do for ATCalsys, since it only has 1 mode. Also, ideally we would make this optional (e.g. `type: CalibrationType | None = None`), and fallback to a "standard" mode for MTCalcys.